hazelcast/src/main/java/com/hazelcast/topic/impl/reliable/ReliableTopicService.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.topic.impl.reliable;
+
+import com.hazelcast.config.ReliableTopicConfig;
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.spi.NodeEngine;
+import com.hazelcast.spi.RemoteService;
+
+public class ReliableTopicService implements RemoteService {
+
+    public static final String SERVICE_NAME = "hz:impl:reliableTopicService";
+
+    private final NodeEngine nodeEngine;
+
+    public ReliableTopicService(NodeEngine nodeEngine) {
+        this.nodeEngine = nodeEngine;
+    }
+
+    @Override
+    public DistributedObject createDistributedObject(String objectName) {
+        ReliableTopicConfig topicConfig = nodeEngine.getConfig().findReliableTopicConfig(objectName);
+        return new ReliableTopicProxy(objectName, nodeEngine, this, topicConfig);
+    }
+
+    @Override
+    public void destroyDistributedObject(String objectName) {
+        //no-op
```

Comment:
I'm not fully understanding the question. There is no client support. I guess you mean a slow consumer?

In case of a listener never coming back ---> the ringbuffer has a fixed capacity so either the item gets overwritten or it expires. A slow consumer can never hang up the system.